dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -202,12 +203,34 @@ func startSyncWriter(q queue.Queue) queue.Queue {
 }
 
 func (cw *ClientWriter) sendSuccessPiece(piece *Piece, cost time.Duration) {
-	cw.api.ReportPiece(piece.SuperNode, &types.ReportPieceRequest{
+	reportPieceRequest := &types.ReportPieceRequest{
 		TaskID:     piece.TaskID,
 		Cid:        cw.cfg.RV.Cid,
 		DstCid:     piece.DstCid,
 		PieceRange: piece.Range,
-	})
+	}
+
+	var retry = 0
+	var maxRetryTime = 3
```

Comment:
The following codes will send request to supernode at most 4 times, is it expected?

Reply:
I don't think so. I use `>=` here `if retry >= maxRetryTime`.

Reply:
Updated.